src/System.Net.Http/src/System/Net/Http/HttpContent.cs

```diff
@@ -309,14 +304,33 @@ public Task CopyToAsync(Stream stream, TransportContext context)
                 throw new ArgumentNullException("stream");
             }
 
-            TaskCompletionSource<object> tcs = new TaskCompletionSource<object>();
+            TaskCompletionSource<object> tcs = new TaskCompletionSource<object>(this);
             try
             {
                 Task task = null;
                 if (IsBuffered)
                 {
-                    byte[] data = this.GetDataBuffer(_bufferedContent);
-                    task = stream.WriteAsync(data, 0, (int)_bufferedContent.Length);
+                    _bufferedContent.Position = 0;
```

Comment:
@stephentoub Good idea! I've added code using `TryGetBuffer` for NETNative. I still think it would be best to use `CopyToAsync` if that isn't available though, since we want to avoid copying the buffer.